src/txmempool.cpp

```diff
@@ -737,7 +738,7 @@ void CTxMemPool::check(const CCoinsViewCache *pcoins) const
             waitingOnDependants.push_back(&(*it));
         else {
             CValidationState state;
-            assert(CheckInputs(tx, state, mempoolDuplicate, false, 0, false, NULL));
+            assert(Consensus::CheckTxInputs(tx, state, mempoolDuplicate, nSpendHeight));
```

Comment:
Hmm, does `CheckInputs` change `state` in `assert` in the original code (i.e. is it against our *Assertions should not have side-effects* rule)?